extras/glusterfs-logrotate

```diff
@@ -61,8 +61,6 @@
     delaycompress
     notifempty
     postrotate
-    for pid in `ps -aef | grep glusterfs | egrep "snapd" | awk '{print $2}'`; do
-        /usr/bin/kill -HUP $pid > /dev/null 2>&1 || true
-    done
+    /usr/bin/killall -HUP `pgrep -f snapd` > /dev/null 2>&1 || true
```

Comment:
Couldn't this return some unexpected processes not belonging to Gluster ? we don't know what else could be running on the same machine, and matching any part of the command line seems too dangerous to me.

Reply:
Is it possible, that there can be two different process running having same name?
If, so how do you suggest we should look out for the process and kill it? Because using the previous method causes the issue as mentioned in the PR description.

Reply:
> Is it possible, that there can be two different process running having same name?

Yes, it's possible but quite unlikely. However the problem here is that "snapd" is not the process name. You are using flag `-f` to search for _snapd_ in any part of the command, including the arguments, which it's where the word will be found.

So it's more likely that another process could be called "snapd" or an argument could contain that word.

> If, so how do you suggest we should look out for the process and kill it? Because using the previous method causes the issue as mentioned in the PR description.

One possibility is to add `grep -v "grep"` to the original command. Another one is to use a more specific pattern for `pgrep`, like `pgrep -f "glusterfs.*snapd"`